spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/AbstractBinder.java

```diff
@@ -212,110 +121,10 @@ public void setCodec(Codec codec) {
 		this.codec = codec;
 	}
 
-	protected IdGenerator getIdGenerator() {
-		return this.idGenerator;
-	}
-
 	public void setIntegrationEvaluationContext(EvaluationContext evaluationContext) {
 		this.evaluationContext = evaluationContext;
 	}
 
-	/**
```

Comment:
The unused methods related to the validation of properties (validateConsumerProperties, validateProducerProperties and validateProperties) can be removed as well.